src/plugins/visualize/public/application/components/deprecation_vis_warning.tsx

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useMemo } from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { EuiCallOut, EuiLink } from '@elastic/eui';
+import { getCoreStart } from '../../services';
+
+export const DeprecationWarning = () => {
+  const documentationLink = useMemo(
+    () => getCoreStart().docLinks.links.visualize.aggregationBased,
+    []
+  );
+
+  return (
+    <EuiCallOut
+      data-test-subj="vizDeprecationWarning"
+      title={
+        <FormattedMessage
+          id="visualize.legacyCharts.notificationMessage"
+          defaultMessage="Deprecated since 7.12, the legacy XY charts will be removed in 7.16. {documentationLink} our new charts library!"
```

Comment:
I think this message is not clear enough, if I were told to "check out" something new I probably wouldn't. Here's my suggestion: `You are using the legacy XY charts library which will be removed in the next minor release, 7.16, after being deprecated in 7.12. Switch to the new library in your advanced settings.`